src/hammer-vlsi/test.py

```diff
@@ -1103,25 +1103,147 @@ def change3(x: hammer_vlsi.HammerTool) -> bool:
 
     def test_resume_pause_hooks_with_custom_steps(self) -> None:
         """Test that resume/pause hooks work with custom steps."""
+        def step5(x: hammer_vlsi.HammerTool) -> bool:
+            with open(os.path.join(c.temp_dir, "step5.txt"), "w") as f:
```

Comment:
To make these a bit easier to reason about could you move them under `with self.create_context() as c:`?

Reply:
If I moved them under `with self.create_context() as c:` then I would have to duplicate the code... I don't see what's hard to reason about here.

Reply:
It's easier to reason about in the sense that `c` appears like it's being used before defined (which it's actually not but had to do a double take here)